src/Badges.sol

```diff
@@ -98,6 +99,43 @@ contract Badges is
     return tokenId;
   }
 
+  function mintWithMerklePermission(
+    address _issuer,
+    bytes32 _merkleRoot,
+    bytes32[] calldata _merkleProof,
+    string calldata _uri,
+    bytes calldata _signature
+  ) external virtual returns (uint256) {
+    require(msg.sender != _issuer, "mintWithMerklePermission: cannot mint if youre the issuer");
+
+    uint256 voucherHashId = safeCheckAgreementMerkle(msg.sender, _issuer, _uri, _signature, _merkleRoot, _merkleProof);
```

Comment:
with additional params needed for expiration, could we try it out? Would we need to use structs?